static/app/components/teamSelector.tsx

```diff
@@ -357,6 +362,23 @@ function TeamSelector(props: Props) {
     [includeUnassigned, multiple, stylesProp]
   );
 
+  useEffect(() => {
+    // Only take action after we've finished loading the teams
+    if (fetching) {
+      return;
+    }
+
+    // If there is only one team, and our flow wants to enable using that team as a default, update the parent state
+    if (options.length === 1 && useTeamDefaultIfOnlyOne) {
+      const castedValue = multiple
+        ? (options as TeamOption[])
+        : (options[0] as TeamOption);
+      handleChange(castedValue);
+    }
+    // We only want to do this once when the component is finished loading for teams and mounted.
+    // If the user decides they do not want the default, we should not add the default value back.
+  }, [fetching, useTeamDefaultIfOnlyOne]);
```

Comment:
When I used all of them, this use effect was happening all the time, and basically you could never clear out the team selection. Or when you did, it would update/render again and the default. Hopefully I'm describing that correctly.